### PR TITLE
Update the Android build documentation

### PR DESCRIPTION
**Pull request by m4gr3d**
- Follow up to https://github.com/godotengine/godot-docs/pull/9579#issuecomment-2246941134
- Follow up to https://github.com/godotengine/godot/issues/94741#issuecomment-2254559168

### CODE REVIEW

File: contributing/development/compiling/compiling_for_android.rst [hunk 1]
```diff
@@ -90,10 +90,10 @@ Setting up the buildsystem
 Building the export templates
 -----------------------------
 
-Godot needs two export templates for Android: the optimized "release"
-template (``android_release.apk``) and the debug template (``android_debug.apk``).
+Godot needs three export templates for Android: the optimized "release"
+template (``android_release.apk``), the debug template (``android_debug.apk``) and the gradle build template (``android_source.zip``).
```
Comment (AThousandShips) [inline contributing/development/compiling/compiling_for_android.rst:94]:
```suggestion
template (``android_release.apk``), the debug template (``android_debug.apk``), and the gradle build template (``android_source.zip``).
```
Please use [Oxford comma](https://docs.godotengine.org/en/latest/contributing/documentation/docs_writing_guidelines.html#use-the-oxford-comma-to-enumerate-anything).

File: contributing/development/compiling/compiling_for_android.rst [hunk 2]
```diff
@@ -164,19 +174,18 @@ You can use the following commands to remove the generated export templates:
...
-When exporting your game, Godot opens the APK, changes a few things inside and
-adds your files.
+When exporting your game, Godot uses the templates as a base, and updates their content as needed.
 
 Installing the templates
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 The newly-compiled templates (``android_debug.apk``
-and ``android_release.apk``) must be copied to Godot's templates folder
+, ``android_release.apk`` and ``android_source.zip``) must be copied to Godot's templates folder
```
Comment (AThousandShips) [inline contributing/development/compiling/compiling_for_android.rst:188]:
```suggestion
, ``android_release.apk`` and, ``android_source.zip``) must be copied to Godot's templates folder
```

File: contributing/development/compiling/compiling_for_android.rst [hunk 3]
```diff
@@ -214,16 +223,20 @@ root directory with the following arguments:
...
+- You can skip certain architectures depending on your target device to speed up compilation. 
+
+Remember to add ``generate_apk=yes`` to the *last* architecture you're building, so that binaries are generated after the build.
 
-The resulting APK will be located at ``bin/android_editor_builds/android_editor-release.apk``.
+The resulting binaries will be located under ``bin/android_editor_builds/``.
 
-Removing the Editor templates
+Removing the Editor binaries
 -----------------------------
```
Comment (AThousandShips) [inline contributing/development/compiling/compiling_for_android.rst:237]:
```suggestion
----------------------------
```

**Review by AThousandShips — APPROVED** [review]

File: contributing/development/compiling/compiling_for_android.rst [hunk 4]
```diff
@@ -90,10 +90,10 @@ Setting up the buildsystem
 Building the export templates
 -----------------------------
 
-Godot needs two export templates for Android: the optimized "release"
-template (``android_release.apk``) and the debug template (``android_debug.apk``).
+Godot needs three export templates for Android: the optimized "release"
+template (``android_release.apk``), the debug template (``android_debug.apk``), and the gradle build template (``android_source.zip``).
```
Comment (Calinou) [inline contributing/development/compiling/compiling_for_android.rst:94]:
```suggestion
Godot needs three export templates for Android: the optimized "release"
template (``android_release.apk``), the debug template (``android_debug.apk``), and the Gradle build template (``android_source.zip``).
```

File: contributing/development/compiling/compiling_for_android.rst [hunk 5]
```diff
@@ -105,22 +106,32 @@ root directory with the following arguments:
...
+
+    scons platform=android target=template_debug arch=arm32 dev_build=yes
+    scons platform=android target=template_debug arch=arm64 dev_build=yes generate_apk=yes
+
+The resulting templates will be located under the ``bin`` directory:
+
+- ``bin/android_release.apk`` for the release template
+- ``bin/android_debug.apk`` for the debug template
+- ``bin/android_dev.apk`` for the dev template
+- ``bin/android_source.zip`` for the gradle build template
```
Comment (Calinou) [inline contributing/development/compiling/compiling_for_android.rst:128]:
```suggestion
- ``bin/android_source.zip`` for the Gradle build template
```

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**